test/test_integration_resultset.rb

```diff
@@ -117,6 +117,13 @@ def test_each
     assert_equal 2, called
   end
 
+  def test_each_enum
+    called = 0
+    @result.reset(1, 2)
+    @result.each.to_a.each { |row| called += 1 }
+    assert_equal 2, called
+  end
```

Comment:
I would like this test to also assert that the `.each` returns an `Enumerator` object. I'd also like a similar test for `.each_hash`, something like:

```ruby
  def test_each_enum
    @result.reset(1, 2)
    enum = @result.each
    assert_instance_of Enumerator, enum
    assert_equal 2, enum.to_a.length
  end

  def test_each_hash_enum
    @result.reset(1, 2)
    enum = @result.each_hash
    assert_instance_of Enumerator, enum
    assert_equal 2, enum.to_a.length
  end
```